src/analyze/tableau/outlier_detection/README.md

```diff
@@ -0,0 +1,26 @@
+Program outlier_detection_ilf.py process Caliper JSON format data collected from Problem Roulette (PR). Refer to https://github.com/tl-its-umich-edu/caliper-event-analysis/tree/master/src/extract/ for scripts that extract data from Problem Roulette's database.
+
+The program should be excecuted from command line. You will need Python version 3.6+ and packages [pandas, math, sklearn, argparse, json] to run the program.
+
+The program use Isolation Forest algorithm to detect rare ones among all PR problem-dealing events and consider them "outliers". Dimensions used in the to detect outliers include:
+
+	IsComplete: whether PR user completed or skipped a problem;
+	IsCorrect: whether PR user's answer to problem was coorect;
+	AttemptCount: number of times PR user tried with the same problem (including current one);
+	AttemptDuration: length of time PR user spent on the problem (measured in seconds).
+
+Command-line options:
+	
+	--perc: Determine percentage of events in data to be considered "outliers". Default value 0.05.
+	--uselog: Decide whether or not to use log value of AttemptDuration. Default value True.
+
+Sample command:
+	
+	python3 --perc 0.1 --uselog f outlier_detection_ilf.py prdata.json
```

Comment:
Some questions about the `--uselog` option:

1. Did you run the program using `--uselog f`?
  I think it won't work as expected, because `argparse` doesn't parse "f" as `False`.  If the argument to a Boolean option contains **_anything_**, then it means `True`.

1. Is the user required to always specify this option?

1. Will the option work if it's not given any "t" or "f" argument?

I recommend treating this option as a Boolean flag instead, without an argument.  Because it's called "uselog", I expect if it's given, then the `log()` function will be used.  If it's not given, then the default behavior would be to not use `log()`.  That will avoid the problem of parsing a Boolean argument.

In the program, use:

```python
parser.add_argument('--uselog', dest='use_log', action='store_true', default=False, help='Use log value of Attempt Duration.')
```

Note that `type` can't be used with `action='store_true'` and `help` is shortened and doesn't show the default value.